model/src/main/resources/reference.yml

```diff
@@ -167,6 +167,9 @@ ort:
 
         timeout: 60
 
+        'map:https://my-repo.example.org(?<repoPath>.*)': "ssh://my-mapped-repo.example.org${repoPath}"
```

Comment:
From a pure user perspective, I believe this reads quite ugly. Can't we have nicely named keys and define the mapping in the values only?

Reply:
Yes, this is certainly not nice. Do you have a concrete proposal?

The key could be built from a prefix plus an arbitrary logical name, e.g. `mapHostXYZ`. I am unsure about the value though. We would need a unique separator between the regex and the replacement that should not appear in one of those.

Reply:
> Yes, this is certainly not nice. Do you have a concrete proposal?

I have only skimmed over the changes, and I'm not sure if I'm getting all use-cases. Is there a use-case where credentials that haven't been part of the URL need to be part of the URL after the mapping? I.e. a simple mapping is not enough?

If so, maybe we could per-define "named groups" / placeholders that don't actually belong to a regex group, but refer to the username / password the authenticator providers for the URL. So you wouldn't have to distinguish between `map` and `mapCred` anymore, and can just use `$username` / `$password` in the replacement URL, as if there were regex groups named like that.

Reply:
I have reworked the URL mapping configuration, so that the regex and the replacement are now defined in the value using ' -> ' as separator. Credentials can now be injected using special variables. Adding credentials to URLs is now handled by URL mappings as well; therefore, the corresponding flag was removed from the FossId configuration. As this changes the configuration, I added the "breaking change" label.